pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -92,9 +92,24 @@ class GraphAgentState:
     retries: int = 0
     run_step: int = 0
 
-    def increment_retries(self, max_result_retries: int, error: BaseException | None = None) -> None:
+    def increment_retries(
+        self, max_result_retries: int, error: BaseException | None = None, max_tokens: int | None = None
+    ) -> None:
         self.retries += 1
         if self.retries > max_result_retries:
+            if (
+                self.message_history
+                and isinstance(model_response := self.message_history[-1], _messages.ModelResponse)
+                and model_response.finish_reason == 'length'
+                and model_response.parts
+                and isinstance(tool_call := model_response.parts[-1], _messages.ToolCallPart)
+            ):
+                try:
+                    tool_call.args_as_dict()
+                except Exception:
+                    raise exceptions.ToolExceedsTokenLimitError(
```

Comment:
I think this should be a subclass of `UnexpectedModelBehavior` for backward compatibility, and I suggest renaming it `IncompleteToolCall`